src/commands/search/kitsu-manga.ts

```diff
@@ -0,0 +1,163 @@
+import type { Kitsu } from '#lib/apis/kitsu/Kitsu';
+import { fetchKitsuApi } from '#lib/apis/kitsu/kitsu-constants';
+import { LanguageKeys } from '#lib/i18n/LanguageKeys';
+import { RedisKeys } from '#lib/redis-cache/RedisCacheClient';
+import { apply } from '#lib/utilities/add-builder-localizations';
+import { bold, EmbedBuilder, hideLinkEmbed, hyperlink, time, TimestampStyles } from '@discordjs/builders';
+import { cutText, isNullishOrEmpty } from '@sapphire/utilities';
+import { AutocompleteInteractionArguments, Command, RegisterCommand } from '@skyra/http-framework';
+import { getSupportedUserLanguageT, resolveUserKey } from '@skyra/http-framework-i18n';
+import { APIApplicationCommandOptionChoice, MessageFlags } from 'discord-api-types/v10';
+
+@RegisterCommand((builder) =>
+	apply(builder, LanguageKeys.Commands.Kitsu.Manga.RootName, LanguageKeys.Commands.Kitsu.Manga.RootDescription) //
+		.addStringOption((option) =>
+			apply(option, LanguageKeys.Common.OptionsManga) //
+				.setRequired(true)
+				.setAutocomplete(true)
+		)
+)
+export class UserCommand extends Command {
+	public override async autocompleteRun(_: Command.AutocompleteInteraction, options: AutocompleteInteractionArguments<Options>) {
+		if (options.focused !== 'manga' || isNullishOrEmpty(options.manga)) {
+			return this.autocompleteNoResults();
+		}
+
+		const result = await fetchKitsuApi('manga', options.manga);
+
+		return result.match({
+			err: () => this.autocompleteNoResults(),
+			ok: async (value) => {
+				const redisInsertPromises: Promise<'OK'>[] = [];
+				const results: APIApplicationCommandOptionChoice[] = [];
+
+				for (const [index, hit] of value.hits?.entries() ?? []) {
+					redisInsertPromises.push(
+						this.container.redisCache.insertFor60Seconds<Kitsu.KitsuHit>(RedisKeys.KitsuManga, options.manga, index.toString(), hit)
+					);
+
+					results.push({
+						name: cutText(hit.titles.en || hit.titles.en_jp || hit.canonicalTitle || hit.slug, 100),
+						value: `${RedisKeys.KitsuManga}:${options.manga}:${index}`
+					});
+				}
+
+				if (redisInsertPromises.length) {
+					await Promise.all(redisInsertPromises);
+				}
+
+				return this.autocomplete({
+					choices: results.slice(0, 19)
+				});
+			}
+		});
+	}
+
+	public override async chatInputRun(interaction: Command.Interaction, { manga }: Options) {
+		const [, packageFromAutocomplete, nthResult] = manga.split(':');
+		const hitFromRedisCache = await this.container.redisCache.fetch<Kitsu.KitsuHit>(RedisKeys.KitsuManga, packageFromAutocomplete, nthResult);
+
+		if (hitFromRedisCache) {
+			return this.buildResponse(hitFromRedisCache, interaction);
+		}
+
+		const result = await fetchKitsuApi('manga', packageFromAutocomplete ?? manga, 1);
+
+		return result.match({
+			err: () => this.handleError(interaction),
+			ok: (value) => {
+				if (!value.hits?.[0]) {
+					return this.handleError(interaction);
+				}
+
+				return this.buildResponse(value.hits[0], interaction);
```

Comment:
Combined to the suggestion above, this can use a ternary, and drop the braces as well.

Reply:
On a side note, and just a nitpick, I prefer the `ok` callback above `err`, feels more natural TBH.